src/main/java/io/nats/streaming/StreamingConnectionImpl.java

```diff
@@ -231,11 +245,24 @@ public void close() throws IOException, InterruptedException {
                 }
                 ackTimer.cancel();
 
+                if (this.messageDispatcher != null && this.messageDispatcher.isActive()) {
+                    nc.closeDispatcher(this.messageDispatcher);
```

Comment:
for the other dispatchers (ack and hb), you unsubscribe, but not here. Is there a reason?